### PR TITLE
[INLONG-985][Agent] Add English Agent binary protocol doc 

### LINKED ISSUE #985: [Improve][Agent] Add Agent binary protocol doc
### Description

Add Agent binary protocol doc

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
Fixes #985 

### Motivation

In order to deepen everyone's understanding of the binary protocol of agent

### Modifications

 Add English Agent binary protocol doc 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
    - *Added integration tests for end-to-end deployment with large payloads (10MB)*
    - *Extended integration test for recovery after broker failure*

### Documentation

No doc needed

### CODE REVIEW

**Review by doleyzi — APPROVED** [review]

**Comment by aloyszhang** [comment]
please add the raw file of image and add `_category_.json` file for directory `docs/deployment/binary_protocol`

File: docs/development/binary_protocol/agent.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+title: Agent binary protocol
+sidebar_position: 2
+---
+## Overview
+This article mainly introduces the flow of Agent data between submodules and the corresponding binary protocol.
+## Data flow
+![](docs/development/binary_protocol/img/agent_1.png)
```
Comment (aloyszhang) [inline docs/development/binary_protocol/agent.md:8]: broken link

Comment (aloyszhang) [reply docs/development/binary_protocol/agent.md:8]: please check all other links too

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by doleyzi — APPROVED** [review]

**Pull request merged**